GitHub Token scanning view (#8563)
pypa/warehouse: warehouse/templates/email/token-compromised-leak/body.txt

{#
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
 # You may obtain a copy of the License at
 #
 # http://www.apache.org/licenses/LICENSE-2.0
 #
 # Unless required by applicable law or agreed to in writing, software
 # distributed under the License is distributed on an "AS IS" BASIS,
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-#}
{% extends "email/_base/body.txt" %}

{% block content %}
What happened?

{% set site_name=request.registry.settings["site.name"] %}

A third party ({{ origin }}) has informed us that a {{ site_name }} API token
associated with your account "{{ username }}" was found to be accessible at the

> following public url. We have automatically revoked this token.

Reviewer: ```suggestion
following public URL. We have automatically revoked this token.
```

{#
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
 # You may obtain a copy of the License at
 #
 # http://www.apache.org/licenses/LICENSE-2.0
 #
 # Unless required by applicable law or agreed to in writing, software
 # distributed under the License is distributed on an "AS IS" BASIS,
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-#}
{% extends "email/_base/body.txt" %}

{% block content %}
What happened?

{% set site_name=request.registry.settings["site.name"] %}

A third party ({{ origin }}) has informed us that a {{ site_name }} API token
associated with your account "{{ username }}" was found to be accessible at the

> following public url. We have revoked this token.

Reviewer: ```suggestion
following public url. We have automatically revoked this token.
```

{{ public_url }}

What should I do?

First, we recommend that you inspect the public page where the token
appeared. Please investigate whether this is the result of an error on your side or if
someone else acquired and published your token.

If the token was not published by you, it means it was stolen, either from

> you or from {{ site_name }}. We urge you to take all the necessary steps to secure all your
> credentials.

Reviewer: ```suggestion
you or from someone with access to your account on {{ site_name }}. We urge you to take all the
necessary steps to secure your credentials.
```

> In case of doubt, you can send an email to admin@pypi.org to communicate with the PyPI
> administrators.
> 

Reviewer: ```suggestion
```

This is already at the end.

> Once you have taken the applicable steps to ensure the token will not leak again,

Reviewer: ```suggestion
Once you have taken the applicable steps to ensure the token will not become public again,
```

you're welcome to generate a new one. Log into your account and head to your profile
page to create a new API token. You will need to update your existing integrations
with the new token.

Lastly, please review the releases of all your packages. You should make
sure a third party did not use your publicly accessible token to impersonate
you and release a malicious package.

How do you know this?

This is an automated message. Our partner {{ origin }} analyzes all the

> contents it recieves for potential {{ site_name }} token leaks and warns us every time

Reviewer: ```suggestion
data it receives for unintentional {{ site_name }} token publications and warns us every time
```
